website_and_docs/content/documentation/webdriver/bidi_apis.en.md

```diff
@@ -89,30 +126,208 @@ ensure
 end
   {{< /tab >}}
   {{< tab header="JavaScript" >}}
-let {Builder, By} = require('selenium-webdriver');
-driver = new Builder().forBrowser('firefox').build();
-
-(async function test(){
-
-    const cdpConnection = await driver.createCDPConnection('page')
-    await driver.logMutationEvents(cdpConnection, function (event) {
-        assert.strictEqual(event['attribute_name'], 'style')
-        assert.strictEqual(event['current_value'], '')
-        assert.strictEqual(event['old_value'], 'display:none;')
-    })
-
-    await driver.get("https://www.google.com")
-
-    let element = driver.findElement({ id: 'reveal' })
-    await element.click()
-    let revealed = driver.findElement({ id: 'revealed' })
-    await driver.wait(until.elementIsVisible(revealed), 5000)
-
-})();
+# Please raise a PR to add code sample
   {{< /tab >}}
   {{< tab header="Kotlin" >}}
 val driver = FirefoxDriver()
 driver.get("http://www.google.com")
 // Please help us create an example for Kotlin
   {{< /tab >}}
 {{< /tabpane >}}
+
+## Listen to console.log events
+
+Listen to the `console.log` events and register callbacks to process the event.
+
+{{< tabpane langEqualsHeader=true >}}
+{{< tab header="Java" >}}
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.devtools.DevTools;
+import org.openqa.selenium.devtools.v87.log.Log;
+
+public void consoleLogTest() {
+ChromeDriver driver = new ChromeDriver();
+DevTools devTools = driver.getDevTools();
+devTools.createSession();
+
+    devTools.send(Log.enable());
+    devTools.addListener(Log.entryAdded(),
```

Comment:
devTools.getDomains().log().entryAdded()

Reply:
Where does the lambda go, then?

Reply:
```java
devTools.addListener(
  devTools.getDomains().log().entryAdded(),
  logEntry -> {
    System.out.println("log: %s", logEntry.getText());
  });
```